### PR TITLE
Use vitest and test WebSocket browser client

### PR DESCRIPTION
**Pull request by ngbrown**
Blocks #68

Includes #70, so merge that first.

This pull request:

- Switches to testing using RabbitMQ and websocket-tcp-relay containers hosted by `docker compose`.
- Switches test framework to `vitest`.
- Adds tests for the WebSocket browser client, and tests with a headless browser in CI. Currently only Chromium but Firefox and Webkit are also options.
- Add new overload for `AMQPWebSocketClient` constructor to allow setting optional parameters through an init object.

The following websocket browser tests are skipped because they cause an unhandled exception and abort the remaining tests:

- can handle frames split over socket reads
- can republish in consume block without race condition

These skipped tests consistently raise errors like the following during test and will be fixed in #68:

```
RangeError: offset is out of bounds
 ❯ AMQPWebSocketClient.handleMessage http:/localhost:63315/src/amqp-websocket-client.ts:78:28
```

```
RangeError: Invalid typed array length: -1251
 ❯ AMQPWebSocketClient.handleMessage http:/localhost:63315/src/amqp-websocket-client.ts:101:28
```

---

Vitest links

* https://vitest.dev/
* https://github.com/vitest-dev/vitest

### CODE REVIEW

**Comment by dentarg** [comment]
> Includes #70, so merge that first.

It has been merged now

File: test-browser/websocket.ts [hunk 1]
```diff
@@ -0,0 +1,670 @@
...
+  const password = 'passwd'
+  const hostname = '127.0.0.1'
+  const port = 15670
+  const vhost = 'my_host'
+  const name = 'test'
+  const client = new AMQPWebSocketClient({ url: `ws://${hostname}:${port}/ws/amqp`, username: username, password: password, vhost: vhost, name: name });
+  expect(client.username).toEqual(username);
+  expect(client.password).toEqual(password);
+  //expect(client.host).toEqual(hostname);
+  //expect(client.port).toEqual(port);
```
Comment (dentarg) [inline test-browser/websocket.ts:21]: I think we can just remove these commented lines

**Review by dentarg — APPROVED** [review]
Nice work, a minor comment

**Comment by dentarg** [comment]
I will note that the `test/test.ts` and `test-browser/websocket.ts` looks very similar, do you think it is possible and worthwhile to share code between them and inject the things that differ (settings, the client object used) or would it get too complex and hard to follow? Thinking about the maintenance going forward. No need for you to address that, just wanted to hear your thoughts about it.

**Comment by dentarg** [comment]
Also, I'll let you rebase this one :)

File: test-browser/websocket.ts (hunk 1 above)
Comment (ngbrown) [reply test-browser/websocket.ts]: Remvoed

**Comment by ngbrown** [comment]
> I will note that the `test/test.ts` and `test-browser/websocket.ts` looks very similar, do you think it is possible and worthwhile to share code between them and inject the things that differ (settings, the client object used) or would it get too complex and hard to follow? Thinking about the maintenance going forward. No need for you to address that, just wanted to hear your thoughts about it.

I added a commit to make the files dramatically more the same, so they are easier to text compare. I added a `getNewClient` generator function that all the tests now use. To compare I use VS Code, select both files, and right-click "Compare Selected". I don't think the two files can be made fully the same because the Node.js specific imports would fail in the browser. There are also a few differences in the tests.

**Comment by dentarg** [comment]
Thanks for this excellent contribution!
